wallet/src/sync.rs

```diff
@@ -262,23 +266,30 @@ async fn apply_transaction<F: Fn(&OuterVerifier) -> bool>(
         .filter(|o| filter(&o.verifier))
         .enumerate()
     {
-        // For now the wallet only supports simple coins, so skip anything else
-        let amount = match output.payload.extract::<Coin<0>>() {
-            Ok(Coin(amount)) => amount,
-            Err(_) => continue,
-        };
-
-        let output_ref = OutputRef {
-            tx_hash,
-            index: index as u32,
-        };
-
-        match output.verifier {
-            OuterVerifier::Sr25519Signature(Sr25519Signature { owner_pubkey }) => {
-                // Add it to the global unspent_outputs table
-                add_unspent_output(db, &output_ref, &owner_pubkey, &amount)?;
+        // For now the wallet only supports simple coins and timestamp
+        match output.payload.type_id {
+            Coin::<0>::TYPE_ID => {
+                let amount = output.payload.extract::<Coin<0>>()?.0;
+
```

Comment:
This may be a good time to make the filtering we do here paramaterizeable like the verifier filter is above. Or at least make a follow up issue for it.